src/lib/support/tests/TestZclString.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements a unit test suite for ZclString
+ *
+ */
+
+#include <inttypes.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <nlunit-test.h>
+#include <support/CHIPMem.h>
+#include <support/CodeUtils.h>
+#include <support/UnitTestRegistration.h>
+
+#include <lib/support/ZclString.h>
+
+using namespace chip;
+using namespace chip::Logging;
+using namespace chip::Platform;
+
+static void TestZclStringWhenBufferIsZero(nlTestSuite * inSuite, void * inContext)
+{
+    uint8_t bufferMemory[1];
+    // Init zcl string
+    MutableByteSpan zclString(bufferMemory);
+    // Init c string
+    char * cString64 = static_cast<char *>(MemoryCalloc(1024, 1));
```

Comment:
Use RAII for allocation (ScopedMemoryBuffer has calloc)

Same for other tests

Reply:
Updated.